File: HospitalManagement/src/main/java/myPack/HospitalManagement/Service/MedicineService.java
package myPack.HospitalManagement.Service;

import java.util.Date;
import java.util.List;

import myPack.HospitalManagement.Entity.Medicine;

public interface MedicineService 
{
	 List<Medicine> getAllMedicines();
	    Medicine getMedicineById(Long id);
	    Medicine saveMedicine(Medicine medicine);
	    Medicine updateMedicine(Long id, Medicine medicine);
	    void deleteMedicine(Long id);
}

File: HospitalManagementDesign/src/app/service/medicine.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import{Observable} from 'rxjs';
import { Medicine } from './medicine';
@Injectable({
  providedIn: 'root'
})
export class MedicineService {
  private baseURL="http://localhost:8080"


  constructor( private httpClient:HttpClient) { }

  getMedicineList():Observable<Medicine[]>{
   return this.httpClient.get<Medicine[]>(`${this.baseURL}/medicines/getAllMedicines`);
    
  }

  createMedicine(medicine:Medicine): Observable<any>{
    return this.httpClient.post(`${this.baseURL}/medicines/postMedicine`, medicine);
  }

  getMedicineById(id:number):Observable<Medicine>{
   return this.httpClient.get<Medicine>(`${this.baseURL}/medicines/${id}`);

  }

  updateMedicine(id: number, medicine: Medicine): Observable<Object> {
    return this.httpClient.put(`${this.baseURL}/medicines/putMedicine/${id}`, medicine);
  }

  deleteMedicine(id:number):Observable<Object>{
    return this.httpClient.delete(`${this.baseURL}/medicines/${id}`);
  }
}
